src/components/NavBar/SearchBarDropdown.tsx

```diff
@@ -130,24 +134,68 @@ interface SearchBarDropdownProps {
   isLoading: boolean
 }
 
-export const SearchBarDropdown = ({
-  toggleOpen,
-  tokens,
-  collections,
+const useSearchTrendingTokens = () => {
+  const { pathname } = useLocation()
+  const isTokenPage = pathname.includes('/tokens')
+  const { data: trendingTokenData } = useTrendingTokens(useWeb3React().chainId)
+  const trendingTokensLength = isTokenPage ? 3 : 2
+  return {
+    data: trendingTokenData,
+    results: useMemo(
+      () => trendingTokenData?.slice(0, trendingTokensLength) ?? [...Array<SearchToken>(trendingTokensLength)],
+      [trendingTokenData, trendingTokensLength]
+    ),
+  }
+}
+
+export const SearchBarDropdown = (props: SearchBarDropdownProps) => {
+  const { isLoading } = props
+  const { chainId } = useWeb3React()
+  const showBNBComingSoonBadge = chainId === SupportedChainId.BNB && !isLoading
+
+  return (
+    <Column overflow="hidden" className={clsx(styles.searchBarDropdownNft, styles.searchBarScrollable)}>
+      <Box opacity={isLoading ? '0.3' : '1'} transition="125">
+        <SuspenseFallbackToPreviousContents>
+          <Suspend when={isLoading}>
+            <SearchBarDropdownContents {...props} />
+          </Suspend>
```

Comment:
Why can't these be unified into a single element? This seems like it is avoiding using the imperative pattern of `useTransition` by setting up a declarative pattern instead. Could you have a composed element that does this all for us (eg `<SuspendFallbackToPreviousContents when={isLoading}>`)? The composed element should also include docs that it allows you to use suspense declaratively instead of imperatively, as through `useTransition`.

Reply:
you could use `SuspendFallbackToPreviousContents` with any component that throws a suspense, and then Suspend is just one component that does so. in fact maybe we should rename Suspened to be more explicit: `SuspendConditionally`, because they are sort of two separate utilities and can both be used independently usefully.

Reply:
tried that out

Reply:
@zzmp made some changes here, lmk if this looks good